### PR TITLE
Fix Travic CI settings for both R and Python

### LINKED ISSUE #84: Refactor travis.yml
I've got `.travis.yml` to work correctly for both R & Python.

It would be a good idea to refactor and update the file however.

For now, the matrix consists of 3 builds: `R (without BUILD_ALL)`, `R (with BUILD_ALL)`, and `Python`
Currently, the two R builds only differ by a single ENV variable: `BUILD_ALL="true"`.
Currently, the Python build uses `python=3.5`.

Here are some todo`s for refactoring:

- [x] Add `python=3.6`, and `python=3.7` to the builds
    - This can be done by preinstalling the different python versions, then creating a pipenv virtualenv specifying the specific python version like the following:
    - `pipenv --python 3.7`
    - Currently, only python 3.5 is being used in the Python build, because I've only managed to get the Travis VM to install python3.5. Figure out a way to install all three versions, then create separate matrix entries for each of python 3.5/3.6/3.7.
- [x] Get rid of all unnecessary lines in `travis.yml`
    - Currently, I may have included unnecessary lines (that don't contribute to the testing) in `travis.yml`. I've got the Travis CI to work correctly, but I'm not sure exactly which lines are helping and which are not. Some testing and cleaning out the code is advised.
- [x] Refactor `travis.yml`
    - Currently, `travis.yml` is very long. It'd be better to move all the action into `Makefile`'s, and just call `make` from `travis.yml`. This is a good reference for how to do that: https://github.com/CCS-Lab/easyml/blob/master/.travis.yml
    - BTW, the original `Makefile` for hbayesdm-py repo is in directory `Python/`. (Although, it is currently not being called to from `travis.yml.`)
- [x] Check if the linting and documenting for py package is working correctly
    - Not sure if the linting:
        - `pipenv run flake8 adopy --format=pylint --statistics --exit-zero`
        - `pipenv run pylint adopy --rcfile=setup.cfg --exit-zero`
    - and documenting:
        - `pipenv run travis-sphinx build`
        - `pipenv run travis-sphinx deploy`
    - are working correctly. Please check the results-log from a Travis CI build (Ex. https://travis-ci.org/CCS-Lab/hBayesDM/jobs/552423723) and see if the linting and documenting is working correctly. (Or if it should even be included in `travis.yml` in the first place.)
- [x] Change the names of the builds for better readability
    - Suggestion: `Sync test`, `R test`, `R test (BUILD_ALL)`, `Python test (3.5)`, `Python test (3.6)`, `Python test (3.7)`

### PR DESCRIPTION
**Pull request by JaeyeongYang**
Closes #84.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@ language: cpp
 # NOTE: was `false`
```
Comment (dlemfh) [inline .travis.yml]: @JaeyeongYang 
This memo line is no longer necessary

**Comment by dlemfh** [comment]
@JaeyeongYang 
If no longer using `pipenv` please also remove pipenv related lines in `REAMDE.md` & `Python/README.rst`

File: Python/hbayesdm/base.py [hunk 2]
```diff
@@ -597,24 +597,32 @@ def _designate_stan_model(self, model: str) -> StanModel:
...
+        if os.path.exists(cache_file):
+            try:
+                with open(cache_file, 'rb') as cached_stan_model:
+                    sm = pickle.load(cached_stan_model)
+                with open(model_path, 'r') as model_stan_code:
+                    assert sm.model_code == model_stan_code.read()
+                does_exist = True
+            except Exception:
+                print('Invalid cached StanModel:', cache_file)
+                print('Remove the cached model...')
```
Comment (dlemfh) [inline Python/hbayesdm/base.py]: I think there'd be less confusion to tell the user: `Removing existing cache model to create a new one...` or `Rewriting cache model...` or sth like that, so as not to confuse user to have to manually "remove the cached model"

**Comment by JaeyeongYang** [comment]
Thanks for suggestions! But This PR is already merged, if you want to make changes, you’d be better to open a separate PR.

**Comment by dlemfh** [comment]
I can't do git operations `git pull` etc. at the moment, why not apply them in the `release/` branch to come? :D

**Comment by JaeyeongYang** [comment]
Release branches are for finalizing and releasing new versions, so we’d better to keep working on the develop or feature branches.

**Pull request merged**